src/libsync/discoveryphase.cpp

```diff
@@ -247,9 +247,13 @@ void DiscoverySingleLocalDirectoryJob::run() {
         emit finishedFatalError(errorString);
         return;
     }
-    errno = 0;
+
     QVector<LocalInfo> results;
-    while (auto dirent = csync_vio_local_readdir(dh, _vfs)) {
+    while (true) {
+        errno = 0;
+        auto dirent = csync_vio_local_readdir(dh, _vfs);
+        if (!dirent)
```

Comment:
nitpick: We actually should check the errno here.  This only works now because there is only one break.